astropy/table/index.py

```diff
@@ -782,18 +782,18 @@ def __init__(self, table):
         if len(self.indices) == 0:
             raise ValueError("Cannot create TableLoc object with no indices")
 
-    def __getitem__(self, item):
+    def __get_row(self, item):
```

Comment:
Do not use double underscores as you want to be able to use this method from the child class. 
Also it returns row**s**, so `_get_rows` seems better.